### PR TITLE
Create functionality to like an article

### PR DESCRIPTION
**Pull request by samdiano**
#### What does this PR do?
Creates endpoint to like an article
#### Description of Task to be completed?
- create like model
- create like controller
- creat routes to like and dislike an article

Have these endpoints working.
`POST /api/v1/articles/:slug/like` 
`POST /api/v1/articles/:slug/disllike` 

#### How should this be manually tested?
After cloning the repo, cd into it and RUN `npm install`, set env variables, and run migration
- Using Postman, test the endpoints above.
- To test authentication, add `x-access-token: JWT TOKEN`. Get the TOKEN from signup endpoint
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160451505

### CODE REVIEW

File: server/controllers/articleController.js [hunk 1]
```diff
@@ -57,12 +67,16 @@ export default class ArticleController {
...
-      where: { slug: req.params.slug }
+      where: { slug: req.params.slug },
+      include: [{
+        model: Like,
+        as: 'likes'
+      }]
     }).then((article) => {
       if (article === null) {
-        res.status(404).json({ message: 'article does not exist', status: 'failed' });
+        res.status(404).json({ message: 'article does not exist', success: 'false' });
```
Comment (samdiano) [inline server/controllers/articleController.js:77]: I changed the response object to match the team's response convention

File: server/controllers/articleController.js [hunk 2]
```diff
@@ -30,8 +30,14 @@ export default class ArticleController {
   * @returns {object} - status, message and list of articles
   */
   static getAllArticles(req, res) {
-    Article.findAll({ limit: 10 })
-      .then(articles => res.status(200).json({ message: 'articles retrieved successfully', status: 'success', articles }))
+    Article.findAll({
+      limit: 10,
+      include: [{
```
Comment (samdiano) [inline server/controllers/articleController.js:35]: I included the likes of an article when fetching the articles

File: server/routes/api/article.js [hunk 3]
```diff
@@ -11,8 +12,9 @@ const router = express.Router();
 router.post('/articles', checkToken, upload.single('imgUrl'), ArticleValidations.validateCreateArticle, ArticleController.createArticle);
 router.get('/articles/feed', ArticleController.getAllArticles);
 router.get('/articles', checkToken, ArticleController.getUserArticles);
-router.get('/articles/:slug', checkToken, ArticleController.getSingleArticle);
+router.get('/articles/:slug', ArticleController.getSingleArticle);
```
Comment (samdiano) [inline server/routes/api/article.js]: I removed the checkToken middleware to enable unauthenticated users to view an article.

**Review by onwuvic — COMMENTED** [review]
LGTM. Why is Travis failing?

**Review by slimsolz — COMMENTED** [review]
LGTM but I think you should get your tests to pass

**Comment by samdiano** [comment]
Travis is failing because I just created seeders folder, and it's not in the staging branch.

**Review by udochukwu — COMMENTED** [review]
LGTM

File: server/routes/api/article.js [hunk 4]
```diff
@@ -17,5 +18,6 @@ router.get('/articles/:slug', ArticleController.getSingleArticle);
 router.delete('/articles/:slug', checkToken, ArticleController.deleteArticle);
 router.put('/articles/:slug', checkToken, upload.single('imgUrl'), ArticleValidations.validateUpdateArticle, ArticleController.updateArticle);
 router.post('/articles/:slug/rate', checkToken, RateValidations.validateRating, RateController.rateArticle);
+router.post('/articles/:slug/:likeType', checkToken, LikeController.likeArticle);
```
Comment (s-akinrele) [inline server/routes/api/article.js:21]: I think `:likeType'` can be a query parameter instead. e.g `?query=value&query=value` @samdiano

**Review by s-akinrele — CHANGES_REQUESTED** [review]
I had some comments

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**